seed/php-sdk/unions/src/Types/Types/UnionWithLiteral.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Seed\Types\Types;
+
+use Seed\Core\Json\JsonProperty;
+use Seed\Core\Json\JsonSkip;
+use Seed\Core\Types\DiscriminatedUnion;
+use Seed\Core\Types\Discriminant;
+
+class UnionWithLiteral extends DiscriminatedUnion
+{
+    /**
+     * @var 'base' $base
+     */
+    #[JsonProperty('base')]
+    public string $base;
+
+    /**
+     * @var 'fern'|'_unknown' $type 
+     */
+    #[JsonProperty('type')]
+    #[Discriminant(['fern' => 'fern'])]
+    public string $type;
+
+    /**
+     * @var 'fern'|mixed
+     */
+    #[JsonSkip]
+    public mixed $value;
```

Comment:
Does this literal even need to support `mixed`? Thoughts on always just requiring it to be `fern`? It would make interacting with the literal string a lot easier from the user's perspective.

I can see that you're using `mixed` to handle the unknown values from the server (which is great), but I wonder if we should be more strict with literal values to start.

I realize you could apply the same thinking here though - if a server ever starts supporting a new value, we ideally don't want to throw. However, failing hard on this literal value is valuable elsewhere (e.g. undiscriminated unions).

Reply:
I'd imagine this is the role of `asFern()`, right?

Reply:
Actually I think I see what you mean here... I'm going to take another look at how much validation we can comfortably replace with serialization-step logic and come back to whether we can drop this mixed type.